persistence/src/main/scala/com/evolutiongaming/akkaeffect/persistence/EventSourcedActorOf.scala

```diff
@@ -158,30 +156,46 @@ object EventSourcedActorOf {
 
   implicit final private class EventStoreOps[F[_], E](val store: EventStore[F, E]) extends AnyVal {
 
-    def asJournaller(seqNrRef: Ref[F, SeqNr])(implicit F: Monad[F]): Journaller[F, E] = new Journaller[F, E] {
-
-      val append = new Append[F, E] {
-
-        def apply(events: Events[E]): F[F[SeqNr]] =
-          seqNrRef
-            .modify { seqNr0 =>
-              events.mapAccumulate(seqNr0) {
-                case (seqNr0, event) =>
-                  val seqNr1 = seqNr0 + 1
-                  seqNr1 -> EventStore.Event(event, seqNr1)
+    def asJournaller(actorCtx: ActorCtx[F], seqNr: SeqNr)(implicit F: Concurrent[F], log: Log[F]): F[Journaller[F, E]] =
+      for {
+        seqNrRef <- Ref[F].of(seqNr)
+      } yield new Journaller[F, E] {
+        val append = new Append[F, E] {
+
+          def apply(events0: Events[E]): F[F[SeqNr]] =
+            seqNrRef
+              .modify { seqNr =>
+                events0.mapAccumulate(seqNr) {
+                  case (seqNr0, event) =>
+                    val seqNr1 = seqNr0 + 1
+                    seqNr1 -> EventStore.Event(event, seqNr1)
+                }
+              }
+              .flatMap { events =>
+                def handleError(err: Throwable) = {
+                  val from = events.values.head.head.seqNr
+                  val to   = events.values.last.last.seqNr
+                  stopActor(from, to, err)
+                }
+                store
+                  .save(events)
+                  .onError(handleError)
+                  .flatTap(_.onError(handleError))
```

Comment:
why there is need for `.flatTap` and error handling?

Reply:
Because I'd like to handle error on both outer and inner effects of type
```
F[F[A]]
```